File: app/src/main/java/com/example/weizhenbin/wangebug/base/AppStatusListener.java
package com.example.weizhenbin.wangebug.base;

import android.app.Activity;
import android.os.Bundle;

/**
 * Created by weizhenbin on 2018/9/18.
 */

public abstract class AppStatusListener extends AppActivityLifecycleListener {
    protected abstract void onAppForeground();
    protected abstract void onAppBackground();


    @Override
    protected void onActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    protected void onActivityStarted(Activity activity) {

    }

    @Override
    protected void onActivityResumed(Activity activity) {

    }

    @Override
    protected void onActivityPaused(Activity activity) {

    }

    @Override
    public void onActivityStopped(Activity activity) {

    }

    @Override
    protected void onActivitySaveInstanceState(Activity activity, Bundle outState) {

    }

    @Override
    protected void onActivityDestroyed(Activity activity) {

    }
}

File: app/src/main/java/com/example/weizhenbin/wangebug/base/AppStatusListener.kt
package com.example.weizhenbin.wangebug.base

import android.app.Activity
import android.os.Bundle

/**
 * Created by weizhenbin on 2018/9/18.
 */

abstract class AppStatusListener : AppActivityLifecycleListener() {
    override fun onActivityCreated(activity: Activity?, savedInstanceState: Bundle?) {
    }

    override fun onActivityStarted(activity: Activity?) {
    }

    override fun onActivityResumed(activity: Activity?) {
    }

    override fun onActivityPaused(activity: Activity?) {
    }

    override fun onActivityStopped(activity: Activity?) {
    }

    override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) {
    }

    override fun onActivityDestroyed(activity: Activity?) {
    }

    override fun onAppForeground() {
    }

    override fun onAppBackground() {
    }
}
